File: 15 Caterpillar method/MinAbsSumOfTwo/minAbsSumOfTwo.js
function solution(A) {
  A.sort((a, b) => a - b);
  const test = [];
  let start = 0;
  let end = A.length - 1;
  let minAbsSum = Math.abs(A[start] + A[end]);
  while (start <= end) {
    const currentSum = A[start] + A[end];
    test.push(`$ {start},$ {end}`);
    if (currentSum == 0) return 0;
    minAbsSum = Math.min(minAbsSum, Math.abs(currentSum));
    if (currentSum <= 0) {
      start += 1;
    } else {
      end -= 1;
    }
  }
  return minAbsSum;
}

File: 15 Caterpillar method/MinAbsSumOfTwo/minAbsSumOfTwo.java
import java.util.Arrays;

class Solution {
  public int solution(int[] A) {
    Arrays.sort(A);
    int start = 0;
    int end = A.length - 1;
    int currentSum = 0;
    int minAbsSum = Math.abs(A[start] + A[end]);
    while (start <= end) {
      currentSum = A[start] + A[end];
      if (currentSum == 0) return 0;
      minAbsSum = Math.min(minAbsSum, Math.abs(currentSum));
      if (currentSum <= 0) {
        start += 1;
      } else {
        end -= 1;
      }
    }
    return minAbsSum;
  }
}
